src/extension/php_objects.c

```diff
@@ -85,26 +102,47 @@ void dd_phpobj_reg_ini_env(const dd_ini_setting *sett)
             .value = env_def ? ZSTR_VAL(env_def) : sett->default_value,
             .value_length =
                 env_def ? ZSTR_LEN(env_def) : (uint32_t)sett->default_value_len,
-
             .on_modify = _on_modify_wrapper,
             .mh_arg1 = (void *)(uintptr_t)sett->field_offset,
             .mh_arg2 = sett->global_variable,
             .mh_arg3 = ZSTR_VAL(entry_ex_fake_str),
         },
         {0}};
 
-    dd_phpobj_reg_ini(defs);
+    if (dd_phpobj_reg_ini(defs) == dd_success &&
+        registered_entries_count < DD_MAX_REGISTERED_ENTRIES) {
+        registered_entries[registered_entries_count].name = name;
+        registered_entries[registered_entries_count].env_name = env_name;
+        registered_entries_count++;
+    } else {
+        if (name) {
+            pefree(name, 1);
+            name = NULL;
+        }
+
+        if (env_name) {
+            pefree(env_name, 1);
+            env_name = NULL;
+        }
+    }
 
     if (env_def) {
         zend_string_efree(env_def);
+        env_def = NULL;
     }
-    efree(name);
 }
 
-static zend_string *nullable _fetch_from_env(const char *name, size_t name_len)
+/*
+ * This function gets call at minit, watch out with using Zend Memory Manager
+ * No emallocs should be used. Instead pemallocs and similars.
+ */
+static char *_get_env_name_from_ini_name(const char *name, size_t name_len)
 {
     size_t env_name_len = ENV_NAME_PREFIX_LEN + name_len;
-    char *env_name = emalloc(env_name_len + 1);
+    char *env_name = pemalloc(env_name_len + 1, 1);
```

Comment:
You can use `safe_pemalloc`, also it doesn't return `NULL`.

Reply:
I'm keeping this vars as `char *` so if I understood correctly, this pemalloc is still correct

Reply:
Well, we know for a fact that `name_len` is unlikely to be of `SIZE_MAX`, so it's probably not strictly necessary.

However you could potentially use `safe_pemalloc` as follows to include overflow checks:
```
safe_pemalloc(env_name_len, 1 /* sizeof(char) */, ENV_NAME_PREFIX_LEN + 1 /* offset */, 1 /* persistent */)

Reply:
Thanks! Done